example/advection/advection_package.cpp

```diff
@@ -38,16 +39,99 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   Real cfl = pin->GetOrAddReal("Advection", "cfl", 0.45);
   pkg->AddParam<>("cfl", cfl);
   Real vx = pin->GetOrAddReal("Advection", "vx", 1.0);
-  pkg->AddParam<>("vx", vx);
   Real vy = pin->GetOrAddReal("Advection", "vy", 1.0);
-  pkg->AddParam<>("vy", vy);
   Real vz = pin->GetOrAddReal("Advection", "vz", 1.0);
-  pkg->AddParam<>("vz", vz);
   Real refine_tol = pin->GetOrAddReal("Advection", "refine_tol", 0.3);
   pkg->AddParam<>("refine_tol", refine_tol);
   Real derefine_tol = pin->GetOrAddReal("Advection", "derefine_tol", 0.03);
   pkg->AddParam<>("derefine_tol", derefine_tol);
 
+  auto profile_str = pin->GetOrAddString("Advection", "profile", "wave");
+  int profile = -1; // unspecified profile
+  if (profile_str.compare("wave") == 0) {
+    profile = 0;
+  } else if (profile_str.compare("smooth_gaussian") == 0) {
+    profile = 1;
+  } else if (profile_str.compare("hard_sphere") == 0) {
+    profile = 2;
+  } else {
+    PARTHENON_FAIL("Unknown profile in advection example: " + profile_str);
+  }
+  pkg->AddParam<>("profile", profile);
```

Comment:
Ah I see what you're up to here. Looks good overall. Can we name these profile variables though? Maybe with an enum or an enum class?
```C++
enum class InitialProfile { unknown = -1; wave = 0; gaussian = 1; sphere = 2; };
```

Reply:
The current version is now directly using the string and no intermediate variable (maybe the browser cached an old version of the PR as this is also already updated to master).

Reply:
Oops my bad. Yeah I must have been reviewing an older version.